app/controllers/results_controller.rb

```diff
@@ -29,6 +30,17 @@ def index
           address4
           postcode
         ]).all
+    @subjects =
+      if params["subjects"].present?
+        SubjectArea.includes(:subjects).all
+          .map(&:subjects).flatten
+          .select { |subject| rails_to_csharp_subject_id(id: subject.id).in?(params["subjects"].split(",").map(&:to_s)) }
```

Comment:
This line is quite verbose and might benefit from being broken out into a seperate method